From dfea37d005b2d4ca02f64e483eef92fb4f0cba7f Mon Sep 17 00:00:00 2001
From: Jaroslav Kysela <jkysela@redhat.com>
Date: Wed, 3 Sep 2014 07:40:01 -0400
Subject: [alsa] usb-audio: Adjust Gamecom 780 volume level

Message-id: <140972823573-405-patchset-jkysela@redhat.com>
Patchwork-id: 91101
O-Subject: [RHEL 7.1 PATCH 404/442] ALSA: usb-audio: Adjust Gamecom 780 volume level
Bugzilla: 1112200
RH-Acked-by: John Linville <linville@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Ivan Vecera <ivecera@redhat.com>

Original patch fixed the original problem, but the sound was far too low
for most users. This patch references a compare matrix to allow the
volume levels to act normally. I personally tested this patch myself,
and volume levels returned to normal. Please see this discussion for
more details: https://bugzilla.kernel.org/show_bug.cgi?id=65251

Signed-off-by: Paul S McSpadden <fisch602@gmail.com>
Cc: <stable@vger.kernel.org> [v3.14+]
Signed-off-by: Takashi Iwai <tiwai@suse.de>

(picked from kernel: 542baf94ec3c5526955b4c9fd899c7f30fae4ebe)
rhbz#1112200
Signed-off-by: Jaroslav Kysela <jkysela@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/sound/usb/quirks.c b/sound/usb/quirks.c
index 7c57f22..19a921e 100644
--- a/sound/usb/quirks.c
+++ b/sound/usb/quirks.c
@@ -670,7 +670,7 @@ static int snd_usb_gamecon780_boot_quirk(struct usb_device *dev)
  /* set the initial volume and don't change; other values are either
   * too loud or silent due to firmware bug (bko#65251)
   */
- u8 buf[2] = { 0x74, 0xdc };
+ u8 buf[2] = { 0x74, 0xe3 };
  return snd_usb_ctl_msg(dev, usb_sndctrlpipe(dev, 0), UAC_SET_CUR,
    USB_RECIP_INTERFACE | USB_TYPE_CLASS | USB_DIR_OUT,
    UAC_FU_VOLUME << 8, 9 << 8, buf, 2);
-- 
1.7.1